onedal/decomposition/incremental_pca.py

```diff
@@ -0,0 +1,99 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+
+from daal4py.sklearn._utils import get_dtype
+
+from ..datatypes import _convert_to_supported, from_table, to_table
+from ..utils import _check_array
+from .pca import BasePCA
+
+
+class IncrementalPCA(BasePCA):
+
+    def __init__(
+        self,
+        n_components=None,
+        is_deterministic=True,
+        method="cov",
+        batch_size=None,
+        whiten=False,
+    ):
+        self.n_components = n_components
+        self.method = method
+        self.is_deterministic = is_deterministic
+        self.batch_size = batch_size
+        self.whiten = whiten
+        module = self._get_backend("decomposition", "dim_reduction")
```

Comment:
it is set wherever backend is called. it is necessary to call partial_result from backend there to initialize correctly. calling _reset is not possible there because _reset is assuming that components are set